core/org.wso2.carbon.utils/src/main/java/org/wso2/carbon/keystore/persistence/dao/KeyStoreDAO.java

```diff
@@ -0,0 +1,272 @@
+import static org.wso2.carbon.keystore.persistence.constant.PersistenceManagerConstants.KeyStoreTableColumns.VERSION;
+import static org.wso2.carbon.keystore.persistence.constant.PersistenceManagerConstants.SqlQueries.ADD_KEY_STORE;
+import static org.wso2.carbon.keystore.persistence.constant.PersistenceManagerConstants.SqlQueries.DELETE_KEY_STORE_BY_NAME;
+import static org.wso2.carbon.keystore.persistence.constant.PersistenceManagerConstants.SqlQueries.GET_ENCRYPTED_KEY_STORE_PASSWORD;
+import static org.wso2.carbon.keystore.persistence.constant.PersistenceManagerConstants.SqlQueries.GET_ENCRYPTED_PRIVATE_KEY_PASSWORD;
+import static org.wso2.carbon.keystore.persistence.constant.PersistenceManagerConstants.SqlQueries.GET_KEY_STORES;
+import static org.wso2.carbon.keystore.persistence.constant.PersistenceManagerConstants.SqlQueries.GET_KEY_STORES_EXISTENCE;
+import static org.wso2.carbon.keystore.persistence.constant.PersistenceManagerConstants.SqlQueries.GET_KEY_STORE_BY_NAME;
+import static org.wso2.carbon.keystore.persistence.constant.PersistenceManagerConstants.SqlQueries.GET_KEY_STORE_LAST_UPDATED_TIME;
+import static org.wso2.carbon.keystore.persistence.constant.PersistenceManagerConstants.SqlQueries.UPDATE_KEY_STORE_BY_NAME;
+import static java.time.ZoneOffset.UTC;
+
+public class KeyStoreDAO {
+
+    private static final Calendar calendar = Calendar.getInstance(TimeZone.getTimeZone(UTC));
+
+    /**
+     * Add a new key store.
+     *
+     * @param keyStoreModel Key store model.
+     */
+    public void addKeystore(KeyStoreModel keyStoreModel) {
+
+        NamedJdbcTemplate namedJdbcTemplate = DatabaseUtil.getNewNamedJdbcTemplate();
+        try {
+            namedJdbcTemplate.executeInsert(ADD_KEY_STORE, (preparedStatement -> {
+                preparedStatement.setString(NAME, keyStoreModel.getName());
+                preparedStatement.setString(TYPE, keyStoreModel.getType());
+                preparedStatement.setString(PROVIDER, keyStoreModel.getProvider());
+                preparedStatement.setString(PASSWORD, keyStoreModel.getEncryptedPassword());
+                preparedStatement.setString(PRIVATE_KEY_ALIAS, keyStoreModel.getPrivateKeyAlias());
+                preparedStatement.setString(PRIVATE_KEY_PASS, keyStoreModel.getEncryptedPrivateKeyPass());
+                preparedStatement.setBinaryStream(CONTENT, new ByteArrayInputStream(keyStoreModel.getContent()),
+                        keyStoreModel.getContent().length);
+                preparedStatement.setString(PUB_CERT_ID, keyStoreModel.getPublicCertId());
+                preparedStatement.setInt(TENANT_ID, keyStoreModel.getTenantId());
+                preparedStatement.setString(VERSION, DEFAULT_VERSION);
+                preparedStatement.setTimeStamp(CREATED_AT, new Timestamp(new Date().getTime()), calendar);
+                preparedStatement.setTimeStamp(UPDATED_AT, new Timestamp(new Date().getTime()), calendar);
```

Comment:
Better to extract `new Timestamp(new Date().getTime())` to a variable and use the same value(/variable) in the both places, rather calculating twice.